### ISSUE TITLE
Having trouble installing on Centos 7

### ISSUE THREAD

**Issue by matthew-page** [wavedrom/wavedrom.github.io#54]
I just downloaded the linux tarball for WaveDrom 2.2.0 and the instructions say to just run the 'wavedrom-editor' and it's giving me a host of errors.  Looking to see if there's an installation guide or some advice to follow?

% ./wavedrom-editor
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
[2788:2788:1205/144944.572976:ERROR:component_loader.cc(165)] Failed to parse extension manifest.
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
[2788:2821:1205/144944.632003:ERROR:bus.cc(393)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
[2818:2818:1205/144944.636854:ERROR:gl_surface_glx.cc(78)] XGetWindowAttributes failed for window 12582916.
[2818:2818:1205/144944.636893:ERROR:gl_surface_glx.cc(839)] Failed to get GLXConfig
[2818:2818:1205/144944.636900:ERROR:gpu_info_collector.cc(60)] gl::GLContext::CreateOffscreenGLSurface failed
[2818:2818:1205/144944.636905:ERROR:gpu_info_collector.cc(199)] Could not create surface for info collection.
[2818:2818:1205/144944.636911:ERROR:gpu_init.cc(66)] gpu::CollectGraphicsInfo failed.
[2840:2840:1205/144944.643461:FATAL:nw_content_renderer_hooks.cc(53)] Failed to load node library (error: libatomic.so.1: cannot open shared object file: No such file or directory)
#0 0x7f730eadfe29 <unknown>

[2818:2818:1205/144944.699357:ERROR:viz_main_impl.cc(176)] Exiting GPU process due to errors during initialization

**Comment by drom** [comment]
Have you tried previous versions?
v2.1.8 or v2.1.2

**Comment by matthew-page** [comment]
no, not yet...does it expect that i've done a full NodeJS support or is 
the tarball "batteries-included" ?

matt

**Comment by drom** [comment]
Yes, each version comes with own runtime, and different, so I can track down the issue

**Comment by matthew-page** [comment]
ok, so if it should be "all included" then which prior version do you suggest I try out?

**Comment by pferguson83** [comment]
I'm have a very similar problem with centos7.3. If there was a dependency list that would be great!

**Comment by drom** [comment]
Please trying v2.1.2

**Comment by pferguson83** [comment]
That worked!!! I guess i have no chance of the latest and greatest release?
is there anything relative to node.js that locks things to a particular machine? I have things installed on an NFS share between a few machines.

Many many thanks for your help!

Here is the output of stdout etc:
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
[5252:5252:0520/093507.181992:ERROR:process_singleton_posix.cc(353)] The profile appears to be in use by another NW.js process (3286) on another computer (*****). NW.js has locked the profile so that it doesn't get corrupted. If you are sure that no other processes are using this profile, you can unlock the profile and relaunch NW.js.
[5252:5252:0520/093507.206401:ERROR:desktop_window_tree_host_x11.cc(1125)] Not implemented reached in virtual void views::DesktopWindowTreeHostX11::InitModalType(ui::ModalType)
ATTENTION: default value of option force_s3tc_enable overridden by environment.
ATTENTION: default value of option force_s3tc_enable overridden by environment.
[5252:5252:0520/093518.678236:ERROR:component_loader.cc(167)] Failed to parse extension manifest.
[5252:5564:0520/093518.772972:ERROR:database.cc(1713)] Web sqlite error 1, errno 0: no such column: type, sql: SELECT card_number_encrypted, last_four,masked.id,metadata.use_count,metadata.use_date,network,type,status,name_on_card,exp_month,exp_year,metadata.billing_address_id,bank_name FROM masked_credit_cards masked LEFT OUTER JOIN unmasked_credit_cards USING (id) LEFT OUTER JOIN server_card_metadata metadata USING (id)

**Comment by drom** [comment]
@pferguson83 
v2.1.2 was bundled with NWJS v0.38.3 = Chromium 74, Node.js v12.2.0
Nothing prevents me from building new WaveDrom 2.6.6 with that version of NWJS
Maybe I should call it Centos7 release ;)

**Comment by arnonsha** [comment]
That would be most helpful. Centos 7 is very common in the EDA industry
